app/server/src/features/accounts/methods.ts

```diff
@@ -64,6 +64,36 @@ export async function findOrgsByUserId(userId: string, prisma: PrismaClient) {
     return formattedData;
 }
 
+/**
+ * get events that this particular user is a moderator of, or has been invited to
+ */
+export async function findEventsByUserId(userId: string, prisma: PrismaClient) {
+    const results = await prisma.user.findUnique({
+        where: { id: userId },
+        include: {
+            moderatorOf: {
+                include: {
+                    event: true,
+                }
+            },
+            invitedOf: {
+                include: {
+                    event: true,
+                }
+            }
+        }
+    });
+
+    if (!results) return null;
+
+    // prepare data for graphql layer
+    const userIsModerator = results.moderatorOf.map(({ event }) => event);
+    const userIsInvited = results.invitedOf.map(({ event }) => event);
+    const formattedData = userIsModerator.concat(userIsInvited);
+
+    return formattedData;
```

Comment:
From my understanding, this could have duplicate events, right?  If someone is both invited and a moderator of a particular event?

Personally, I think the function might be doing too much. It would make sense to me if on the dashboard there's a "events I'm moderating" section. Then this could be split into two functions `getInvitedTo` and `getModeratorOf` with two fields on the graphql model maybe `invitedTo` and `moderatorOf`.

What do you think?

From a user perspective I think it looks more wrong for the same event to appear twice in the same section as opposed to once in one section and once in another section.

Reply:
To clarify, would we like to see the events organized on the dashboard first by Ongoing/Upcoming, then by Moderator/Invited? Or only by Moderator/Invited?